src/modules/directory.rs

```diff
@@ -32,8 +32,25 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         .config_value_i64("fish_style_pwd_dir_length")
         .unwrap_or(FISH_STYLE_PWD_DIR_LENGTH);
 
+    // Using environment PWD is the standard approach for determining logical path
+    let use_logical_path = module.config_value_bool("use_logical_path").unwrap_or(true);
+    let logical_current_dir = if use_logical_path {
```

Comment:
Do all shells update `$PWD`?
I think that resorting to hiding the directory if `$PWD` is unavailable could make it difficult for users to fix the problem. What are your thoughts around falling back to the physical path rather than returning `None`?

Reply:
I can't guarantee that all shells update PWD, but the GNU Coreutils `pwd` specifically says in its manpage that it works by trying to read the value of the envar `PWD`. I figure if it's good enough for coreutils, it's good enough for us.

Reply:
As far as falling back to physical path goes: that's actually the intention of this code, so I guess I need to add some comments to make that clear. If the logical path is unreadable, we log an error and return None (this is just so that the user will have a clue as to why the behavior has changed).

Further down, `None` values are converted into the physical path (or at least, the path returned by `std::env::current_dir`, which might not actually be the physical path).

The total effect of this is that if `use_logical_path` is false OR `PWD` is unreadable, the prompt will fall back to the physical path.